cpp/tests/io/json_test.cpp

```diff
@@ -666,4 +716,151 @@ TEST_F(JsonReaderTest, StringInference)
+    "18446744073709551613", "18446744073709551614", "18446744073709551615", "-5"};
+  auto input_out_of_range_positive =
+    column_wrapper<cudf::string_view>(out_of_range_positive.begin(), out_of_range_positive.end());
+  auto input_out_of_range_negative =
+    column_wrapper<cudf::string_view>(out_of_range_negative.begin(), out_of_range_negative.end());
+  auto input_greater_uint64_max =
+    column_wrapper<cudf::string_view>(greater_uint64_max.begin(), greater_uint64_max.end());
+  auto input_less_int64_min =
+    column_wrapper<cudf::string_view>(less_int64_min.begin(), less_int64_min.end());
+  auto input_mixed_range =
+    column_wrapper<cudf::string_view>(mixed_range.begin(), mixed_range.end());
+
+  auto out_of_range_positive_append_zeros = append_zeros(out_of_range_positive, 32, true);
+  auto out_of_range_negative_append_zeros = append_zeros(out_of_range_negative, 5);
+  auto greater_uint64_max_append_zeros    = append_zeros(greater_uint64_max, 8, true);
+  auto less_int64_min_append_zeros        = append_zeros(less_int64_min, 17);
+  auto mixed_range_append_zeros           = append_zeros(mixed_range, 2, true);
+
+  auto input_out_of_range_positive_append = column_wrapper<cudf::string_view>(
+    out_of_range_positive_append_zeros.begin(), out_of_range_positive_append_zeros.end());
+  auto input_out_of_range_negative_append = column_wrapper<cudf::string_view>(
+    out_of_range_negative_append_zeros.begin(), out_of_range_negative_append_zeros.end());
+  auto input_greater_uint64_max_append = column_wrapper<cudf::string_view>(
+    greater_uint64_max_append_zeros.begin(), greater_uint64_max_append_zeros.end());
+  auto input_less_int64_min_append = column_wrapper<cudf::string_view>(
+    less_int64_min_append_zeros.begin(), less_int64_min_append_zeros.end());
+  auto input_mixed_range_append = column_wrapper<cudf::string_view>(
+    mixed_range_append_zeros.begin(), mixed_range_append_zeros.end());
+
+  auto filepath = temp_env->get_temp_dir() + "ParseOutOfRangeIntegers.json";
+  {
+    std::ostringstream line;
+    for (int i = 0; i < num_rows; ++i) {
+      line << "[" << out_of_range_positive[i] << "," << out_of_range_negative[i] << ","
+           << greater_uint64_max[i] << "," << less_int64_min[i] << "," << mixed_range[i] << ","
+           << out_of_range_positive_append_zeros[i] << "," << out_of_range_negative_append_zeros[i]
+           << "," << greater_uint64_max_append_zeros[i] << "," << less_int64_min_append_zeros[i]
+           << "," << mixed_range_append_zeros[i] << "]\n";
+    }
+    std::ofstream outfile(filepath, std::ofstream::out);
+    outfile << line.str();
+  }
+  cudf_io::json_reader_options in_options =
+    cudf_io::json_reader_options::builder(cudf_io::source_info{filepath}).lines(true);
+
+  cudf_io::table_with_metadata result = cudf_io::read_json(in_options);
+
+  const auto view = result.tbl->view();
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(input_out_of_range_positive, view.column(0));
```

Comment:
Can this be CUDF_TEST_EXPECT_COLUMNS_EQUAL ?

Reply:
Yes, these lines in `TEST_F(JsonReaderTest, ParseOutOfRangeIntegers)`
 can be made `CUDF_TEST_EXPECT_COLUMNS_EQUAL`, however the ones in `TEST_F(JsonReaderTest, ParseInRangeIntegers)` cannot because the input column and the column from the table read from the file do not match up on their `nullable` property.

I can either keep that comparison as equivalent or add validity in the test input column to enforce equality of `nullable` property.

Reply:
That's fine, now that I look at it, _EQUIVALENT performs all relevant comparisons for this test.